shakenfist/daemons/queues.py

```diff
@@ -265,7 +265,9 @@ def instance_delete(instance_uuid, new_state, error_msg=None):
         instance = virt.Instance.from_db(instance_uuid)
         if instance:
             instance.delete()
-            instance.update_state(new_state, error_message=error_msg)
+            instance.update_state(new_state)
+            if error_msg:
+                instance.error_msg = error_msg
```

Comment:
If update_state() set the error message for error states, this would not need to change. In fact we don't do it now, but should entering a state other than error ensure that the error message in the database is blank?